### PR TITLE
test: Edge cases - Token Service Tests Part 1

### LINKED ISSUE #19757: Edge cases - Token Service Tests

### PR DESCRIPTION
**Pull request by vtronkov**
**Description**:
* Part 1 of 3 to add token edge cases wrapped in an atomic batch

**Related issue(s)**:

Fixes #19757

### CODE REVIEW

**Review by mhess-swl — APPROVED** [review]

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/token/batch/AtomicTokenAssociationSpecs.java [hunk 1]
```diff
@@ -0,0 +1,717 @@
...
+        final String treasury2 = "TREASURY_2";
+        return hapiTest(
+                overridingTwo("tokens.maxPerAccount", "1", "entities.limitTokenAssociations", "true"),
+                cryptoCreate(alice),
+                cryptoCreate(TOKEN_TREASURY),
+                cryptoCreate(treasury2),
+                tokenCreate(VANILLA_TOKEN).treasury(TOKEN_TREASURY),
+                tokenCreate(KNOWABLE_TOKEN).treasury(treasury2),
+                tokenAssociate(alice, KNOWABLE_TOKEN),
+                atomicBatch(tokenAssociate(alice, VANILLA_TOKEN)
```
Comment (mhess-swl) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/token/batch/AtomicTokenAssociationSpecs.java:219]: I see a few tests that only use one transaction inside of the batch. I doubt this will be very common in production–why not include more transactions in the batch? In this case, the `tokenAssociate` above?

Comment (vtronkov) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/token/batch/AtomicTokenAssociationSpecs.java:219]: @mhess-swl How I see it is that everything up to the `tokenAssociate` in the batch is the test setup. I can include more ops in the atomicBatch(in this case, the first `tokenAssociate`), but I think it's not part of what we want to test in this particular test. I understand that having one op in an atomic batch is not very practical, but it will show us that if there is an error, in this case `TOKENS_PER_ACCOUNT_LIMIT_EXCEEDED`, we will handle it correctly. And to test that it won't matter if the two `tokenAssociate` ops are in the atomicBatch.
Especially when we have an op that throws an error, we are rolling back everything anyway, and it won't have any effect.

Comment (mhess-swl) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/token/batch/AtomicTokenAssociationSpecs.java:219]: Eh, testing the _aggregated_ effects of multiple inner transactions (in the same batch) is much more valuable imo than testing the effects of a test batch transaction that only has a single inner transaction, but it doesn't need to hold up the PR

**Review by mhess-swl — APPROVED** [review]

**Review by mhess-swl — APPROVED** [review]

**Comment by vtronkov** [comment]
Closing due to DCO issue. New PR: https://github.com/hiero-ledger/hiero-consensus-node/pull/19900
